src/modules/common/components/Headroom.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import raf from 'raf'
+
+import style from './Headroom.css'
+
+class Headroom extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.any.isRequired,
+    scrollHeight: PropTypes.number.isRequired,
+    pinStart: PropTypes.number.isRequired,
+    stickyAncestor: PropTypes.node,
+    height: PropTypes.number
+  }
+
+  static defaultProps = {
+    pinStart: 0
+  }
+
+  constructor (props) {
+    super(props)
+    this.handleEvent = this.handleEvent.bind(this)
+    this.update = this.update.bind(this)
+    this.ticking = false
+    this.lastScrollY = 0
+    this.lastTransform = 0
+    this.lastKnownScrollY = 0
+    this.currentScrollY = 0
+    this.direction = 'down'
+    this.state = {transform: 0}
+  }
+
+  componentDidMount () {
+    window.addEventListener('scroll', this.handleEvent)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll touchstart touchmove', this.handleEvent)
+  }
+
+  getScrollY = () => {
+    if (window.pageYOffset !== undefined) {
+      return window.pageYOffset
+    } else if (window.scrollTop !== undefined) {
+      return window.scrollTop
+    } else {
+      return (document.documentElement || document.body.parentNode || document.body).scrollTop
+    }
+  }
+
+  truncate (transform, currentScrollY) {
+    if (transform + currentScrollY < this.props.pinStart) {
+      return 0
+    }
+    return Math.min(0, Math.max(-this.props.scrollHeight, transform))
+  }
+
+  update () {
+    // this.lastScrollY: ScrollTop when the scroll direction last changed
+    // this.lastTransform: Transform when the scroll direction last changed
+    // this.lastKnownScrollY: Last ScrollTop (to detect change in scroll direction)
+    // this.currentScrollY: Current ScrollTop
+    // this.direction: 'up' || 'down'
+
+    this.currentScrollY = this.getScrollY()
+    if (this.lastKnownScrollY < this.currentScrollY) { // We're moving up
+      if (this.direction !== 'up') { // We've changed direction from down to up!
+        this.lastTransform = this.state.transform
+        this.lastScrollY = this.lastKnownScrollY
+        this.direction = 'down'
```

Comment:
shouldn't this be: `this.direction = 'up'`?

Reply:
No, but my comment(s) are wrong ^^

Reply:
but if `this.direction !== 'up'` is true then direction is already down

Reply:
Ok that's funny :D why does it work then? :D